app/src/main/java/org/thunderdog/challegram/ui/MainController.java

```diff
@@ -1573,4 +1953,582 @@ private void checkSyncAlert () {
+      options.info(title);
+    }
+    if (isFolder) {
+      options.item(new OptionItem(R.id.btn_editFolder, Lang.getString(R.string.EditFolder), OPTION_COLOR_NORMAL, R.drawable.baseline_edit_24));
+      int chatFolderStyle = tdlib.settings().chatFolderStyle();
+      if (chatFolderStyle == ChatFolderStyle.ICON_ONLY || chatFolderStyle == ChatFolderStyle.LABEL_AND_ICON) {
+        options.item(new OptionItem(R.id.btn_changeFolderIcon, Lang.getString(R.string.ChatFolderChangeIcon), OPTION_COLOR_NORMAL, R.drawable.baseline_image_24));
+      }
+      options.item(new OptionItem(R.id.btn_folderIncludeChats, Lang.getString(R.string.ChatFolderAddChats), OPTION_COLOR_NORMAL, R.drawable.baseline_add_24));
+    }
+    if (isMain) {
+      options.item(new OptionItem(R.id.btn_hideFolder, Lang.getString(R.string.HideAllChats), OPTION_COLOR_NORMAL, R.drawable.baseline_eye_off_24));
+    } else if (isFolder || isArchive) {
+      options.item(new OptionItem(R.id.btn_hideFolder, Lang.getString(R.string.HideFolder), OPTION_COLOR_NORMAL, R.drawable.baseline_eye_off_24));
+    }
+    if (isFolder) {
+      options.item(new OptionItem(R.id.btn_removeFolder, Lang.getString(R.string.RemoveFolder), OPTION_COLOR_RED, R.drawable.baseline_delete_24));
+    }
+    if (options.itemCount() > 0) {
+      options.item(OptionItem.SEPARATOR);
+    }
+    options.item(new OptionItem(R.id.btn_chatFolders, Lang.getString(R.string.EditFolders), OPTION_COLOR_NORMAL, Build.VERSION.SDK_INT >= Build.VERSION_CODES.N ? R.drawable.baseline_edit_folders :  R.drawable.baseline_rule_folder_24));
+    showOptions(options.build(), (v, id) -> {
+      if (id == R.id.btn_editFolder) {
+        tdlib.send(new TdApi.GetChatFolder(chatFolderId), (result) -> runOnUiThreadOptional(() -> {
+          switch (result.getConstructor()) {
+            case TdApi.ChatFolder.CONSTRUCTOR:
+              TdApi.ChatFolder chatFolder = (TdApi.ChatFolder) result;
+              EditChatFolderController controller = new EditChatFolderController(context, tdlib);
+              controller.setArguments(new EditChatFolderController.Arguments(chatFolderId, chatFolder));
+              navigateTo(controller);
+              break;
+            case TdApi.Error.CONSTRUCTOR:
+              UI.showError(result);
+              break;
+          }
+        }));
+      } else if (id == R.id.btn_hideFolder) {
+        tdlib.settings().setChatListEnabled(chatList, false);
+        if (headerCell != null && !StringUtils.isEmptyOrBlank(title)) {
+          context()
+            .tooltipManager()
+            .builder(headerCell.getTopView())
+            .locate((targetView, outRect) -> outRect.left = outRect.right = Screen.dp(56f) - (int) targetView.getX())
+            .controller(this)
+            .show(tdlib, Lang.getString(R.string.HideFolderInfo, title, Lang.getString(R.string.Settings), Lang.getString(R.string.ChatFolders)))
+            .hideDelayed(3500, TimeUnit.MILLISECONDS);
```

Comment:
```suggestion
          if (isMain && !tdlib.hasPremium()) {
            context()
              .tooltipManager()
              .builder(headerCell.getTopView())
              .locate((targetView, outRect) -> outRect.left = outRect.right = Screen.dp(56f) - (int) targetView.getX())
              .controller(this)
              .show(tdlib, "Subscribe to Telegram Premium to be able to hide \"All Chats\" tab.")
              .hideDelayed(3500, TimeUnit.MILLISECONDS);
          } else {
            context()
              .tooltipManager()
              .builder(headerCell.getTopView())
              .locate((targetView, outRect) -> outRect.left = outRect.right = Screen.dp(56f) - (int) targetView.getX())
              .controller(this)
              .show(tdlib, Lang.getString(R.string.HideFolderInfo, title, Lang.getString(R.string.Settings), Lang.getString(R.string.ChatFolders)))
              .hideDelayed(3500, TimeUnit.MILLISECONDS);
          }
```

Reply:
Added a control that allows premium users to hide the 'All Chats' tab

Reply:
@null-nick The feature of hiding "All chats" was previously available in this PR, however, it's been decided that it's not acceptable feature for the mass audience, as this would make it basically impossible to access chats that do not belong to any folder. For example, you might receive a notification from them, but if you dismiss it, there's no way to open that chat. Therefore it was removed at least until such cases would be properly handled.